distributed/tests/test_client.py

```diff
@@ -3109,6 +3110,36 @@ async def test_submit_on_cancelled_future(c, s, a, b):
         await c.submit(inc, x)
 
 
+@gen_cluster(client=True)
+async def test_compute_partially_forgotten(c, s, *workers):
+    # (CPython impl detail)
+    # While it is not possible to argue about what the iteration order of a set
+    # will be, it is determinisitic and only depends on the hash of the inserted
+    # elements. Therefore, converting the set to a list will alway yield the
+    # same order. We're initializing the keys in this very specific order to
+    # ensure that the scheduler internally arranges the keys in this way
+
+    # We'll need the list to be
+    # ['key', 'lost_dep_of_key']
+    keys = key, lost_dep_of_key = list({"foo", "bar"})
+
+    task = Task(key, inc, TaskRef(lost_dep_of_key))
+    res = c.get({task.key: task}, keys, sync=False)
+
+    res = c.get({task.key: task}, keys, sync=False)
```

Comment:
Only happens if it is submitted once. The first submission leaves a zombie task around after triggering the "lost deps" exception. That zombie causes the second one to trigger the transition error.